src/Core/FeatureFlag/Layer/DotEnvLayer.php

```diff
@@ -0,0 +1,147 @@
+
+namespace PrestaShop\PrestaShop\Core\FeatureFlag\Layer;
+
+use PrestaShop\PrestaShop\Core\EnvironmentInterface;
+use PrestaShop\PrestaShop\Core\FeatureFlag\FeatureFlagSettings;
+use PrestaShop\PrestaShop\Core\FeatureFlag\TypeLayerInterface;
+
+class DotEnvLayer implements TypeLayerInterface
+{
+    public function __construct(
+        private EnvironmentInterface $environment
+    ) {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isReadonly(): bool
+    {
+        // It's always editable via DotEnv layer!
+        return false;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getTypeName(): string
+    {
+        return FeatureFlagSettings::TYPE_DOTENV;
+    }
+
+    /**
+     * Retrieve the variable name of this feature flag.
+     */
+    public function getVarName(string $featureFlagName): string
+    {
+        return FeatureFlagSettings::PREFIX . strtoupper($featureFlagName);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function canBeUsed(string $featureFlagName): bool
+    {
+        return isset($_ENV['SYMFONY_DOTENV_VARS']) &&
+            str_contains($_ENV['SYMFONY_DOTENV_VARS'], $this->getVarName($featureFlagName));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isEnabled(string $featureFlagName): bool
+    {
+        return isset($_ENV[$this->getVarName($featureFlagName)]) &&
+            filter_var($_ENV[$this->getVarName($featureFlagName)], \FILTER_VALIDATE_BOOLEAN);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function enable(string $featureFlagName): void
+    {
+        $this->setStatus($featureFlagName, true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function disable(string $featureFlagName): void
+    {
+        $this->setStatus($featureFlagName, false);
+    }
+
+    /**
+     * Retrieve dotenv file used with this feature flag set.
+     */
+    private function locateDotEnvFile(string $featureFlagName): string
+    {
+        $env = $this->environment->getName();
+        $filesToCheck = [".env.$env.local", ".env.$env", '.env'];
+
+        foreach ($filesToCheck as $file) {
+            $path = _PS_ROOT_DIR_ . '/' . $file;
+            if (file_exists($path) && str_contains(file_get_contents($path), $this->getVarName($featureFlagName))) {
+                return $path;
+            }
+        }
+
+        return '';
+    }
+
+    /**
+     * Set feature flag status in the good dotenv file.
+     */
+    private function setStatus(string $featureFlagName, bool $status): void
+    {
+        if ($pathDotenv = $this->locateDotEnvFile($featureFlagName)) {
+            file_put_contents(
+                $pathDotenv,
+                preg_replace(
+                    "/({$this->getVarName($featureFlagName)})=(.*)/",
+                    "$1={$this->boolLabel($status)}",
+                    file_get_contents($pathDotenv)
+                )
+            );
+
+            return;
+        }
```

Comment:
I'm not sure to understand ???

Reply:
It's aim to edit the value of the feature flag in the dotenv file used.

Reply:
I think there is a few other PHP classes in prestashop that perform this operation "modify a key/value file row". Maybe this can be abstracted

```
private function setStatus(string $featureFlagName, bool $status): void
{
    $this->myAwesomeFileManager->modifyLineValue($this->getVarName($featureFlagName), $this->boolLabel($status));
```

https://github.com/PrestaShop/PrestaShop/blob/develop/src/Adapter/Debug/DebugMode.php#L191

Reply:
This is a very specific code, it's not really a good practice in the first place so moving this into a dedicated util class would incite doing this in other places We don't want to encourage this kind of thing so keeping this only in this context seems safer

Here we allow it only because it can only impact .env variables related to feature flags, since it's based on a prefix it can't modify any other future variable